encoder-otel-brave/src/test/java/zipkin2/reporter/otel/brave/ITZipkinSenderTests.java

```diff
@@ -0,0 +1,223 @@
+import brave.Tracing;
+import brave.handler.MutableSpan;
+import brave.handler.SpanHandler;
+import brave.propagation.ThreadLocalCurrentTraceContext;
+import brave.propagation.TraceContext;
+import brave.sampler.Sampler;
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import okhttp3.OkHttpClient;
+import okhttp3.OkHttpClient.Builder;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import zipkin2.reporter.AsyncReporter;
+import zipkin2.reporter.Encoding;
+import zipkin2.reporter.okhttp3.OkHttpSender;
+
+@Testcontainers(disabledWithoutDocker = true)
+class ITZipkinSenderTests {
+
+  private static final Logger log = LoggerFactory.getLogger(ITZipkinSenderTests.class);
+
+  private static final int EXPECTED_TRACE_SIZE = 5;
+
+  @Container
+  JaegerAllInOne jaegerAllInOne = new JaegerAllInOne();
+
+  JaegerTestingScenario testingScenario;
+
+  @Test
+  void shouldSendBraveSpansToJaegerOtlpEndpoint() throws Exception {
+    testingScenario = new JaegerTestingScenario(
+        jaegerAllInOne.getHttpOtlpPort());
+
+    List<String> traceIds = testingScenario.exportedTraceIds();
+
+    Awaitility.await().untilAsserted(() -> {
+      then(traceIds).hasSize(EXPECTED_TRACE_SIZE);
+      thenAllTraceIdsPresentInBackend(testingScenario.queryUrl(), traceIds);
+    });
+  }
+
+  @AfterEach
+  void shutdown() {
+    testingScenario.close();
+  }
+
+  private static void thenAllTraceIdsPresentInBackend(String queryUrl, List<String> traceIds) {
+    OkHttpClient client = new Builder()
+        .build();
+
+    traceIds.forEach(traceId -> {
+      Request request = new Request.Builder().url(queryUrl + traceId).build();
+      try (Response response = client.newCall(request).execute()) {
+        then(response.isSuccessful()).isTrue();
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    });
+  }
+
+  /**
+   * Sender: Brave OKHttp with OTLP proto over HTTP ; Receiver: Jaeger OTLP
+   */
+  class JaegerTestingScenario {
+
+    private final BraveTraceIdGenerator braveTraceIdGenerator;
+
+    JaegerTestingScenario(int otlpPort) {
+      this.braveTraceIdGenerator = new BraveTraceIdGenerator(otlpPort);
+    }
+
+    String queryUrl() {
+      return "http://localhost:" + jaegerAllInOne.getQueryPort() + "/api/traces/";
+    }
+
+    List<String> exportedTraceIds() throws Exception {
+      return braveTraceIdGenerator.traceIds();
+    }
+
+    public void close() {
+      braveTraceIdGenerator.close();
+    }
+  }
+
+  /**
+   * Actual testing logic that uses Brave to generate spans and send them to the backend.
+   */
+  static class BraveTraceIdGenerator {
```

Comment:
inline this into the test please. As there is no logic that behaves differently with different trace IDs, please use a static trace ID

Reply:
This class is replaced by [ITOtelEncoderTest](https://github.com/openzipkin-contrib/zipkin-otel/pull/7/files#diff-f1614b313ae6e7ebe173556465333def6f893b421571522e9bb496a42ede83c3)